packages/disputer/index.js

```diff
@@ -155,12 +155,8 @@ async function run(
       await delay(Number(pollingDelay));
     }
   } catch (error) {
-    logger.error({
-      at: "Disputer#index",
-      message: "Disputer polling error🚨",
-      error: typeof error === "string" ? new Error(error) : error
-    });
-    await waitForLogger(logger);
+    // If any error is thrown, catch it and bubble up to the main try-catch for error processing in the Poll function.
+    throw typeof error === "string" ? new Error(error) : error;
```

Comment:
im not sure where in the code strings are thrown as errors, but this check is really weird to me.  worth pointing out that with nodes `assert` you can easily generate descriptive errors with less code.

Reply:
Ye, it is a bit janky and strange. We do this in a few places as sometimes there are errors generated by upstream components that come out as strings. Specifically, web3.js can sometimes make errors that are not consistent in type, such as WebSocket errors. Ideally, We should refactor this to be cleaner and more consistent but for now, we've found this to be the most robust way to catch these kinds of issues.